src/library/attribute_manager.h

```diff
@@ -7,65 +7,149 @@ Author: Leonardo de Moura
 #pragma once
 #include "kernel/environment.h"
 #include "library/io_state.h"
+#include "util/priority_queue.h"
 
 #ifndef LEAN_DEFAULT_PRIORITY
 #define LEAN_DEFAULT_PRIORITY 1000u
 #endif
 
 namespace lean {
+
+struct attr_data {
+    virtual unsigned hash() const {
+        return 0;
```

Comment:
You probably want this to be a pure-virtual method: 
`virtual unsigned hash() const = 0`